src/Parser.js

```diff
@@ -33,19 +36,41 @@ class Parser {
   }
 
   registerExtension(ext, parser) {
-    if (!ext.startsWith('.')) {
+    const type = toType(ext);
+
+    if (type === 'regexp' || isRegexp.test(ext)) {
+      ext = ext.toString();
+    } else if (!ext.startsWith('.')) {
       ext = '.' + ext;
     }
 
     this.extensions[ext] = parser;
   }
 
+  // find extension key
+  findExt(ext) {
+    const keys = Object.keys(this.extensions);
+
+    for (let i = keys.length - 1; i >= 0; i--) {
+      const key = keys[i];
+
```

Comment:
Might do the simple check first `ext === key` since it's the cheaper and more common operation.